submariner/pluggable-network-plugin-and-ovn.md

```diff
@@ -0,0 +1,50 @@
+## Add support for OVN with pluggable network-plugin support
+
+[submariner#778](https://github.com/submariner-io/submariner/issues/778)
+
+## Summary
+
+Today submariner supports working with existing network-plugins which are based
+on the kubeproxy and iptables implementations. Any network-plugin outside of this
+case won't work, OVN-Kubernetes is an example.
+
+This is a proposal to support pluggingable network drivers, starting by OVN-kubernetes.
+
+## Proposal
+
+Re-architect submariner-routeagent in a way that different network-plugin implementations
+can be provided. Those drivers would be notified of different transitions on a node:
+
+* Transition of a gateaway node in your cluster
+* Connection of a new cluster (and it's CIDR details)
+* Disconnection of existing clusters (and it's CIDR details)
+* ... any other event necessary -TBD-
+
+In the same way that we have cable-drivers today, the route-agent would become aware
+of the underlying network plugin the current cluster uses.
+
+These network-plugin specific drivers will interface to the underlying network-plugin
+and prepare it's configuration as necessary, as well as any routing or linux network
+stack changes necessary on the worker/gateway nodes of the cluster.
+
+
+#### Technologies explored and alternatives
+
+* Talk about eBPF
```

Comment:
I think we can briefly summarize why we did not consider eBPF and provide a link to the eBPF Analysis document here

Reply:
https://docs.google.com/document/d/1KcrxZROb3v-1EW8B-lOKls4GNwWl73Zg02LytNEbuck/edit